client/auth.go

```diff
@@ -0,0 +1,386 @@
+package client
+
+import (
+	"crypto/rand"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+
+	"github.com/google/uuid"
+	xlineapi "github.com/xline-kv/go-xline/api/xline"
+	"golang.org/x/crypto/pbkdf2"
+)
+
+// Client for Auth operations.
+type authClient struct {
+	// Name of the AuthClient, which will be used in CURP propose id generation
+	name string
+	// The client running the CURP protocol, communicate with all servers.
+	curpClient curpClient
+	// The auth token
+	token string
+}
+
+// Creates a new `AuthClient`
+func newAuthClient(name string, curpClient curpClient, token string) authClient {
+	return authClient{
+		name:       name,
+		curpClient: curpClient,
+		token:      token,
+	}
+}
+
+// Enables authentication.
+func (c *authClient) AuthEnable() (*xlineapi.AuthEnableResponse, error) {
+	requestWithToken := xlineapi.RequestWithToken{
+		Token: &c.token,
+		RequestWrapper: &xlineapi.RequestWithToken_AuthEnableRequest{
+			AuthEnableRequest: &xlineapi.AuthEnableRequest{},
+		},
+	}
+	resp, err := c.handleReq(&requestWithToken, false)
+	if err != nil {
+		return nil, err
+	}
+	res := resp.CommandResp.GetAuthEnableResponse()
+	res.Header.Revision = resp.SyncResp.Revision
+	return res, err
+}
+
+// Disables authentication.
+func (c *authClient) AuthDisable() (*xlineapi.AuthDisableResponse, error) {
+	requestWithToken := xlineapi.RequestWithToken{
+		Token: &c.token,
+		RequestWrapper: &xlineapi.RequestWithToken_AuthDisableRequest{
+			AuthDisableRequest: &xlineapi.AuthDisableRequest{},
+		},
+	}
+	resp, err := c.handleReq(&requestWithToken, false)
+	if err != nil {
+		return nil, err
+	}
+	res := resp.CommandResp.GetAuthDisableResponse()
+	res.Header.Revision = resp.SyncResp.Revision
+	return res, err
+}
+
+// Gets authentication status.
+func (c *authClient) AuthStatus() (*xlineapi.AuthStatusResponse, error) {
+	requestWithToken := xlineapi.RequestWithToken{
+		Token: &c.token,
+		RequestWrapper: &xlineapi.RequestWithToken_AuthStatusRequest{
+			AuthStatusRequest: &xlineapi.AuthStatusRequest{},
+		},
+	}
+	resp, err := c.handleReq(&requestWithToken, true)
+	if err != nil {
+		return nil, err
+	}
+	res := resp.CommandResp.GetAuthStatusResponse()
+	return res, err
+}
+
+func (c *authClient) Authenticate(request *xlineapi.AuthenticateRequest) (*xlineapi.AuthenticateResponse, error) {
+	requestWithToken := xlineapi.RequestWithToken{
+		Token: &c.token,
+		RequestWrapper: &xlineapi.RequestWithToken_AuthenticateRequest{
+			AuthenticateRequest: request,
+		},
+	}
+	resp, err := c.handleReq(&requestWithToken, true)
```

Comment:
This should use slow path. Descibed in https://github.com/xline-kv/Xline/pull/435#discussion_r1311106704

Reply:
Thank you for your feedback. I've updated the pull request based on your suggestions.